pkg/registry/networking/ingressclass/strategy.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingressclass
+
+import (
+	"context"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/networking"
+	"k8s.io/kubernetes/pkg/apis/networking/validation"
+)
+
+// ingressClassStrategy implements verification logic for IngressClass
+// resources.
+type ingressClassStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating
+// IngressClass objects.
+var Strategy = ingressClassStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+// NamespaceScoped returns false because IngresClass is a non-namespaced
+// resource.
+func (ingressClassStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// PrepareForCreate clears the status of an Ingress before creation.
+func (ingressClassStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
```

Comment:
why not  * type (doesn't copy when called)

Reply:
Are you asking about turning the obj parameter into a `*runtime.Object`? That would make sense, but this has to [implement the RESTCreateStrategy interface](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/rest/create.go#L56).

Reply:
s/ingressClassStrategy/*ingressClassStrategy

Reply:
I'm wondering why the methods take the non-pointer type, which incurs a struct copy?

Reply:
Thanks for the clarification! Unfortunately that change breaks things:

```
pkg/registry/networking/ingressclass/storage/storage.go:43:3: cannot use ingressclass.Strategy (type ingressclass.ingressClassStrategy) as type rest.RESTCreateStrategy in field value:
	ingressclass.ingressClassStrategy does not implement rest.RESTCreateStrategy (PrepareForCreate method has pointer receiver)
```